feature/interface/tunnel/otg_tests/tunnel_based_gre_encap/README.md

```diff
@@ -0,0 +1,155 @@
+# TUN-1.1: Interface-based traffic IPv4 GRE Encapsulation using next-hop redirection
+
+## Summary
+
+The test verifies traffic encapsulation using a IPv4 tunnel GRE interface using next-hop redirection via policy forwarding.
+
+## Testbed type
+
+*  [`featureprofiles/topologies/atedut_4.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_4.testbed)
+
+## Procedure
+
+### Test environment setup
+
+*   DUT has an ingress port and 2 egress ports.
+
+    ```
+                             |         | ---- | ATE Port 2 |
+        [ ATE Port 1 ] ----  |   DUT   |      |            |
+                             |         | ---- | ATE Port 3 |
+    ```
+
+*   Traffic is generated from ATE Port 1.
+*   ATE Port 2 is used as the destination port for encapsulated 
+    traffic.
+*   ATE Port 3 is used as the fallback destination for
+    pass-through traffic.
+
+
+#### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces. Configure MTU of 9216 (L2) on ATE Port1, Port2
+ 
+2.  IPv4 and IPv6 static routes to test destination networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  A set of IPv4 static routes to 32x IPv4 GRE tunnel endpoints towards ATE Port 2.
+
+4.  32 tunnel interfaces are configured wtih IPv4 tunnel endpoints as destination and device loopback as source. Configure MTU on tunnel interfaces to 2000.
+    All tunnel interfaces should be unnumbered referencing loopback interface.
+
+5.  IPv4 encapsulation next-hop (IPV4-ENCAP-NH) is allocated. A set of IPv4 static routes to IPV4-ENCAP-NH is configured toward 32 tunnel interfaces.
+
+6.  2 IPv4 and 2 IPv6 source prefixes will be used to generate tests traffic 
+(SRC1-SRC2). Apply policy-forwarding with 4 rules to DUT Port 1:
+    - Match IPV4-SRC1 and accept/foward.
+    - Match IPV6-SRC1 and accept/foward.
+    - Match IPV4-SRC2 and set next-hop to IPV4-ENCAP-NH.
+    - Match IPV6-SRC2 and set next-hop to IPV4-ENCAP-NH.
```

Comment:
The test should specify the OC used for this more explicitly as this is the primary goal of this test and it might be unclear exactly how to use the OC for this.  Based on the intent for this test, I would suggest using stateless GRE encap (ie: not using tunnel interfaces).  


## create a prefix set 
/defined-sets/ipv4-prefix-sets/ipv4-prefix-set
/defined-sets/ipv6-prefix-sets/ipv6-prefix-set

## create a policy to match the prefix set and action to gre-encap

### match prefix set
/routing-policy/policy-definitions/policy-definition/statements/statement/conditions/match-prefix-set/config/prefix-set	

### action encap-gre

/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source	

/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination

Reply:
We should change this test to what I think is the preferred approach of using stateless / policy-forwarding rules as that is the use case here.  

If a vendor cannot support this and requires using tunnel interfaces to achieve the same behavior, a deviation can be implemented in the code as a workaround.